### PR TITLE
AN-1506 device available width and height

### PR DESCRIPTION
**Pull request by KulenovicInes**
### Work
- Fixes: https://bitmovin.atlassian.net/browse/AN-1506
- Description: 

https://developer.android.com/reference/android/view/WindowManager#getDefaultDisplay()
https://developer.android.com/reference/android/view/Display#getMetrics(android.util.DisplayMetrics)

To be able to use `getRealMetrics()` we would need to update minSdkVersion to 17 (android 4.2)

https://developer.android.com/reference/android/view/Display#getRealMetrics(android.util.DisplayMetrics)

In Jira ticket there is a updated table with dimensions that were sent previously, values these methods return and with results of calculations.

As discussed with Roland, we need to include dimensions of status bar, so best approach would be to use real dimensions and divide them with screen density. As can be seen in table, values we get this way are similar to those we would get using `context.resources.configuration`, since that method returns available dimensions without status bar or some other screen decoration elements.

### Checklist

- [ ] Updated CHANGELOG.md

### CODE REVIEW

**Comment by KulenovicInes** [comment]
> So this now gives us the actual "density independent" pixels? I thought that's what the "dp" stuff from Android was about anyways? Why don't we use `screenWidthDp `?

Because eg. `screenHeightDp` does not include status bar height. According to documentation values form those methods don't include "screen decoration elements".  When discussing this with @rolandgriesser we concluded that it would maybe be more accurate measurement if we used real dimensions. 🤔

File: collector/src/main/java/com/bitmovin/analytics/data/DeviceInformationProvider.kt [hunk 1]
```diff
@@ -4,20 +4,32 @@ import android.app.UiModeManager
...
 
 open class DeviceInformationProvider(val context: Context, val userAgent: String) {
     var isTV: Boolean = isTVDevice()
 
     fun getDeviceInformation(): DeviceInformation {
+
+        val windowManager: WindowManager = context.getSystemService(Context.WINDOW_SERVICE) as WindowManager
+        val displayMetrics = DisplayMetrics()
+
+        windowManager.defaultDisplay.getMetrics(displayMetrics)
```
Comment (rolandgriesser) [inline collector/src/main/java/com/bitmovin/analytics/data/DeviceInformationProvider.kt:20]: Can WindowManager be `null`?

File: CHANGELOG.md [hunk 2]
```diff
@@ -7,6 +7,9 @@ and this project adheres to [Semantic Versioning](http://semver.org/).
 
 ## Development
 
+### Changed
+- Way of getting available screen width and height
```
Comment (rolandgriesser) [inline CHANGELOG.md:11]: Should we mention that we changed that to include screen density?

File: collector/src/main/java/com/bitmovin/analytics/data/DeviceInformationProvider.kt (hunk 1 above)
Comment (KulenovicInes) [reply collector/src/main/java/com/bitmovin/analytics/data/DeviceInformationProvider.kt:20]: I believe that `WindowManager` is non nullable type. As long as `context` is not null, everything should be fine. `getSystemService()` method should return null only in cases when some random string is passed as name parameter, and service with that name doesn't exist.

We already use `context.getSystemService()` in `isTVDevice()` method, so I think it should be safe to use it here.

File: CHANGELOG.md (hunk 2 above)
Comment (KulenovicInes) [reply CHANGELOG.md]: Changed in https://github.com/bitmovin/bitmovin-analytics-collector-android/pull/118/commits/ccd9a3fcaab2a09c15624b04ec51bd0cab7e062d. Does this maybe has more sense?

File: collector/src/main/java/com/bitmovin/analytics/data/DeviceInformationProvider.kt (hunk 1 above)
Comment (Tigraine) [reply collector/src/main/java/com/bitmovin/analytics/data/DeviceInformationProvider.kt:20]: probably should get a guard or something here

Comment (KulenovicInes) [reply collector/src/main/java/com/bitmovin/analytics/data/DeviceInformationProvider.kt]: Added null check in https://github.com/bitmovin/bitmovin-analytics-collector-android/pull/118/commits/02c97d8446258f0af46b08139bf4fd3cebc34db6

**Pull request merged**